shell/platform/darwin/ios/ios_external_texture_gl.mm

```diff
@@ -55,10 +66,82 @@
   }
 }
 
+void IOSExternalTextureGL::CreateYUVTexturesFromPixelBuffer() {
+  size_t width = CVPixelBufferGetWidth(buffer_ref_);
+  size_t height = CVPixelBufferGetHeight(buffer_ref_);
+  {
+    CVOpenGLESTextureRef yTexture;
+    CVReturn err = CVOpenGLESTextureCacheCreateTextureFromImage(
+        kCFAllocatorDefault, cache_ref_, buffer_ref_, nullptr, GL_TEXTURE_2D, GL_LUMINANCE, width,
+        height, GL_LUMINANCE, GL_UNSIGNED_BYTE, 0, &yTexture);
+    if (err != noErr) {
+      FML_DCHECK(yTexture) << "Could not create texture from pixel buffer: " << err;
+    } else {
+      y_texture_ref_.Reset(yTexture);
+    }
+  }
+
+  {
+    CVOpenGLESTextureRef uvTexture;
+    CVReturn err = CVOpenGLESTextureCacheCreateTextureFromImage(
+        kCFAllocatorDefault, cache_ref_, buffer_ref_, nullptr, GL_TEXTURE_2D, GL_LUMINANCE_ALPHA,
+        width / 2, height / 2, GL_LUMINANCE_ALPHA, GL_UNSIGNED_BYTE, 1, &uvTexture);
```

Comment:
Why is this indiscriminately using half resolution?  I'm guessing lossy use of these channels is hard to notice?  Seems like any loss should be optional.

Reply:
This code comes from[ Apple’s documentation](https://developer.apple.com/documentation/corevideo/1619860-cvopenglestexturecachecreatetext?language=objc), under "Mapping the chroma plane of a 420v buffer as a source texture".

Reply:
Since 420v is sampled at 2:1 in the vertical and horizontal directions, every four Y components share a set of UV components, so only half the length and width are used